### PR TITLE
killport: init at 0.8.0

### PR DESCRIPTION
**Pull request by SnO2WMaN**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/killport/default.nix
Comment (ocfox) [inline pkgs/tools/misc/killport/default.nix:11]: maybe use inherit?

Comment (SnO2WMaN) [reply pkgs/tools/misc/killport/default.nix]: fixed.

**Comment by IndeedNotJames** [comment]
https://github.com/jkfran/killport/releases/tag/v0.7.0 has been released in the meantime :)

Also, does not build on `aarch64-darwin` and `x86_64-darwin` (see ci checks/ofborg).

**Comment by SnO2WMaN** [comment]
> https://github.com/jkfran/killport/releases/tag/v0.7.0 has been released in the meantime :)
> 
> Also, does not build on `aarch64-darwin` and `x86_64-darwin` (see ci checks/ofborg).

Anyway, I updated to 0.7.0. later I'll check that this can't build on `aarch64-darwin` and `x86_64-darwin`

**Comment by SnO2WMaN** [comment]
> > https://github.com/jkfran/killport/releases/tag/v0.7.0 has been released in the meantime :)
> > Also, does not build on `aarch64-darwin` and `x86_64-darwin` (see ci checks/ofborg).
> 
> Anyway, I updated to 0.7.0. later I'll check that this can't build on `aarch64-darwin` and `x86_64-darwin`

I managed to do something about `aarch64-darwin`, but I couldn't figure out about `x86_64-darwin` (since I don't have a machine with that architecture to begin with), can someone please do something about that?

**Comment by IndeedNotJames** [comment]
Summoning @wegank who knows `darwin` (and because I don't want to ping the whole `darwin-maintainers` team).

But you could *probably* just add `meta.broken = stdenv.isDarwin && stdenv.isAarch64` or even `platforms = platforms.linux;` and call it a day :^)

---

Some minor formating nit:
```patch
diff --git a/pkgs/tools/misc/killport/default.nix b/pkgs/tools/misc/killport/default.nix
index 1c2b3bf884e..ca862ad6323 100644
--- a/pkgs/tools/misc/killport/default.nix
+++ b/pkgs/tools/misc/killport/default.nix
@@ -3,6 +3,7 @@
 , fetchCrate
 , llvmPackages
 }:
+
 rustPlatform.buildRustPackage rec {
   pname = "killport";
   version = "0.7.0";
@@ -20,8 +21,6 @@ rustPlatform.buildRustPackage rec {
     description = "A command-line tool to easily kill processes running on a specified port";
     homepage = "https://github.com/jkfran/killport";
     license = licenses.mit;
-    maintainers = with maintainers; [
-      sno2wman
-    ];
+    maintainers = with maintainers; [ sno2wman ];
   };
```

File: pkgs/top-level/all-packages.nix
Comment (wegank) [inline pkgs/top-level/all-packages.nix:37845]:
```suggestion
  killport = darwin.apple_sdk_11_0.callPackage ../tools/misc/killport { };
```

File: pkgs/tools/misc/killport/default.nix
Comment (wegank) [inline pkgs/tools/misc/killport/default.nix:17]:
```suggestion
  nativeBuildInputs = [ rustPlatform.bindgenHook ];
```

File: pkgs/tools/misc/killport/default.nix
Comment (wegank) [inline pkgs/tools/misc/killport/default.nix:15]:
```suggestion
  cargoHash = "sha256-u5HJLBT8hYKaJQBFaJUb2yyQrmPsN6+P0mTH+iHDPu0=";
```

File: pkgs/tools/misc/killport/default.nix
Comment (wegank) [inline pkgs/tools/misc/killport/default.nix:2]:
```suggestion
{ lib
, rustPlatform
```

File: pkgs/tools/misc/killport/default.nix
Comment (wegank) [reply pkgs/tools/misc/killport/default.nix]: > This suggestion seems to break build on `x86_64-darwin`

Yes, sorry about this. I guess the global SDK bump is really an urgent need on that platform.

File: pkgs/tools/misc/killport/default.nix
Comment (wegank) [inline pkgs/tools/misc/killport/default.nix:2]:
```suggestion
{ lib
, rustPlatform
```

Also this.

**Comment by SnO2WMaN** [comment]
Anyway, I've squashed these commit, and ready to merge.

File: pkgs/tools/misc/killport/default.nix
Comment (IndeedNotJames) [inline pkgs/tools/misc/killport/default.nix:8]:
```suggestion
  version = "0.8.0";
```

v0.8.0 released in the meantime
https://github.com/jkfran/killport/releases/tag/v0.8.0

(don't forget to update the hashes below too)

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/killport/default.nix
Comment (SnO2WMaN) [reply pkgs/tools/misc/killport/default.nix]: I updated and ready to merge(if check passed)

**Comment by wegank** [comment]
@ofborg build killport killport.passthru.tests

**Review by wegank — APPROVED** [review]

**Review by emilylange — APPROVED** [review]

**Pull request merged**
